site/src/pages/SetupPage/SetupPageView.tsx

```diff
@@ -112,12 +137,26 @@ export const SetupPageView: FC<SetupPageViewProps> = ({
 			},
 			validationSchema,
 			onSubmit,
+			// With validate on blur set to true, the form lights up red whenever
+			// you click out of it. This is a bit jarring. We instead validate
+			// on submit and change.
+			validateOnBlur: false,
 		});
 	const getFieldHelpers = getFormHelpers<TypesGen.CreateFirstUserRequest>(
 		form,
 		error,
 	);
 
+	const onEmailChange = useCallback(
+		(event: ChangeEvent<HTMLInputElement>) => {
+			const email = event.target.value;
+			const username = usernameFromEmail(email);
+			form.setFieldValue("username", username);
+			onChangeTrimmed(form)(event);
+		},
+		[form],
+	);
```

Comment:
IMO, there is no need to use `useCallback` since the function is not used as a dependency in an effect and the function doesn't look heavy for optimization.

https://www.joshwcomeau.com/react/usememo-and-usecallback/

Reply:
moved it out of `useCallback`